pkg/api/pod/util_test.go

```diff
@@ -3472,3 +3472,95 @@ func TestDropClusterTrustBundleProjectedVolumes(t *testing.T) {
 		})
 	}
 }
+
+func TestDropPodLifecycleSleepAction(t *testing.T) {
+	makeSleepHandler := func() *api.LifecycleHandler {
+		return &api.LifecycleHandler{
+			Sleep: &api.SleepAction{Seconds: 1},
+		}
+	}
+	makeContainer := func(preStop, postStart *api.LifecycleHandler) api.Container {
+		container := api.Container{Name: "foo"}
+		if preStop != nil || postStart != nil {
+			container.Lifecycle = &api.Lifecycle{
+				PostStart: postStart,
+				PreStop:   preStop,
+			}
+		}
+		return container
+	}
+
+	makePod := func(containers []api.Container) *api.PodSpec {
+		return &api.PodSpec{Containers: containers}
+	}
+
+	testCases := []struct {
+		description string
+		gateEnabled bool
+		oldPod      *api.PodSpec
+		newPod      *api.PodSpec
+		wantPod     *api.PodSpec
+	}{
+		{
+			description: "gate disabled, new pod with action, old pod nil",
```

Comment:
I think you are missing a lot of interesting cases to prove no side-effects:

With gate disabled:
- nil -> nil = nil
- nil -> Exec = Exec
- nil -> Sleep = nil
- Exec -> Exec = Exec
- Exec -> HTTP = HTTP
- Exec -> Sleep = nil
- Sleep -> Exec = Exec
- Sleep -> Sleep = Sleep

Repeat for both preStop and postStart.
Repeat for all 3 container lists
Repeat with gate enabled

Reply:
Updated, tried my best to refactor :)
(Honestly, I don't think we actually need that much cases for `update` (like `Exec -> Http`), `Lifecycle` seems is immutable?